### ISSUE TITLE
manifest authoring or manifest reading is broken for the latest stage 0

### ISSUE THREAD

**Issue by wli3** [dotnet/sdk#19866]
symptom if you build a project with TFM `net6.0;net6.0-android` or `net6.0-ios` all would result in asking to install macos, instead of android or ios.

Further, the following test fails.

```
        [Fact]
        public void ItShouldReturnListOfManifestFiles2()
        {

            var sdkDirectoryWorkloadManifestProvider
                = new SdkDirectoryWorkloadManifestProvider(sdkRootPath: @"C:\work\sdk2\artifacts\bin\redist\Debug\dotnet", sdkVersion: "6.0.100");

            var workloadResolver = WorkloadResolver.Create(sdkDirectoryWorkloadManifestProvider, @"C:\work\sdk2\artifacts\bin\redist\Debug\dotnet", "6.0.100");

            var suggestedWorkloads = workloadResolver.GetWorkloadSuggestionForMissingPacks(
                new List<WorkloadPackId> { new WorkloadPackId("Microsoft.Android.Sdk") },
                out ISet<WorkloadPackId> unsatisfiablePacks
            );

            suggestedWorkloads.Should().NotBeEmpty();
        }
```

![image](https://user-images.githubusercontent.com/6993335/129991433-e03312f8-dd1d-4cca-8495-dddd09a2f7d1.png)

There is only 4 expanded workload
Something wrong with manifests or manifest reader

**Comment by wli3** [comment]
@sfoslund @mhutch could you look into this issue? I have a branch https://github.com/dotnet/sdk/compare/main...wli3:bad-manifest-auth?expand=1 to help repro.

I will work on making build works for now.

**Comment by sfoslund** [comment]
@wli3 I've debugged this and I believe the root cause has to do with the testing configuration. The workloads are all getting correctly read from the manifests but [IsWorkloadPlatformCompatible](https://github.com/dotnet/sdk/blob/865fdb6eb3cc47048b8b0a68ae267a1682c3361c/src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadResolver.cs#L644) is saying that most of the workloads are not compatible with the current RID because they all either directly or indirectly via extends require a certain set of platforms. This should be fine, since one of the valid platforms is win-x64, but the resolver actually failed to figure out what the current runtime identifier is, because it's looking for the file `C:\code\sdk\artifacts\bin\redist\Debug\dotnet\sdk\6.0.100\NETCoreSdkRuntimeIdentifierChain.txt` when the file actually exists at ``C:\code\sdk\artifacts\bin\redist\Debug\dotnet\sdk\6.0.100-dev\NETCoreSdkRuntimeIdentifierChain.txt``.

Long story short, when I change the SDK version in the test from `6.0.100` to `6.0.100-dev`, everything works.

**Comment by wli3** [comment]
I get a similar behavior in dog food env. Should that happen?

**Comment by wli3** [comment]
Build a project with net6.0-android, the error ask me to install macos

**Comment by sfoslund** [comment]
Same problem with dogfood, when the show missing workloads target is called, it's called with the stage 0 sdk version instead of 6.0.100-dev, so the runtime identifier file can't be found: https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.ImportWorkloads.targets#L37

**Comment by wli3** [comment]
ok, so i need to figure out how to pass that in properly. I'll keep this ticket open for now since I disabled several tests due to that
